src/compiler/parser.ts

```diff
@@ -3225,6 +3228,16 @@ namespace ts {
             }
         }
 
+        function parseAssertsTypePredicate(): TypeNode {
+            const node = <TypePredicateNode>createNode(SyntaxKind.TypePredicate);
+            node.assertsModifier = parseExpectedToken(SyntaxKind.AssertsKeyword);

```

Comment:
Is there a possibility that there will be more modifiers in the future? If so, would it make sense to put this into `Node#modifiers`?

Reply:
It's possible, but for now I'm going to keep it the way it is.